src/cascadia/Remoting/Monarch.cpp

```diff
@@ -147,6 +147,56 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
         }
     }
 
+    uint64_t Monarch::_lookupPeasantIdForName(const winrt::hstring& name)
+    {
+        if (name.empty())
+        {
+            return 0;
+        }
+
+        std::vector<uint64_t> peasantsToErase{};
+        uint64_t result = 0;
+        for (const auto& [id, p] : _peasants)
+        {
+            try
+            {
+                auto otherName = p.WindowName();
+                if (otherName == name)

```

Comment:
Case sensitive compare? Are you sure? Lots of people love to be lazy on the capitalization when doing a command line thing and I suspect that might apply to `-w`.